src/expression.rs

```diff
@@ -143,7 +150,17 @@ impl Expression {
                 Some(value) => Number(*value),
                 None => Variable(identifier),
             },
-            _ => self,
+            Address(memory_reference) => {
+                let number = patch_values.and_then(|patch_values| {
```

Comment:
Maybe don't shadow the `patch_values` name inside the `and_then`? Matter of taste more than anything.

Reply:
Yeah I don't like shadowing either but I struggle with what to call `Option<Something>` once I've verified that it's `Some(something)`.

Reply:
How about just `pv`? Small variable names seem less of an issue if it’s referring to the inner thing.